ddsketch/store/dense_store.go

```diff
@@ -267,8 +267,39 @@ func (s *DenseStore) Encode(b *[]byte, t enc.FlagType) {
 	if s.IsEmpty() {
 		return
 	}
-	enc.EncodeFlag(b, enc.NewFlag(t, enc.BinEncodingContiguousCounts))
+
+	denseEncodingSize := 0
 	numBins := uint64(s.maxIndex-s.minIndex) + 1
+	denseEncodingSize += enc.Uvarint64Size(numBins)
+	denseEncodingSize += enc.Varint64Size(int64(s.minIndex))
+	denseEncodingSize += enc.Varint64Size(1)
+
+	sparseEncodingSize := 0
+	numNonEmptyBins := uint64(0)
+
+	previousIndex := s.minIndex
+	for index := s.minIndex; index <= s.maxIndex; index++ {
+		count := s.bins[index-s.offset]
+		countVarFloat64Size := enc.Varfloat64Size(count)
+		denseEncodingSize += countVarFloat64Size
+		if count != 0 {
```

Comment:
>Do you need to do lines 283/284 if count is 0?

Yes, it takes one byte in the dense encoding if so.

>Also it seems like you could use a rougher heuristic here like the number of bins instead of being so precise if you want to get some cycles back, but I think this implementation is fine as well

Yes, I think we can revisit later if need be. Also, we could probably stop iterating early when the sparse encoding size gets too large, given that we can easily calculate upfront the largest possible dense encoding size.